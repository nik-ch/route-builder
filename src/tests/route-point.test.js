import React from "react";
import renderer from "react-test-renderer";

import RoutePoint from "../components/route-point";

test("Renders correctly without geo data", () => {
    const routePoint = renderer.create(<RoutePoint />);
    expect(routePoint.toJSON()).toMatchSnapshot();
});

test("Renders correctly with geo data provided", () => {
    const geoData = { name: "name", description: "description" };
    const routePoint = renderer.create(<RoutePoint geoDataShown={true} geoData={geoData} />)
    expect(routePoint.toJSON()).toMatchSnapshot();
});

test("Renders correctly when drag started", () => {
    const routePoint = renderer.create(<RoutePoint dragStarted={true}/>);
    expect(routePoint.toJSON()).toMatchSnapshot();
});

test("Renders correctly when drag entered", () => {
    const routePoint = renderer.create(<RoutePoint dragEntered={true}/>);
    expect(routePoint.toJSON()).toMatchSnapshot();
});