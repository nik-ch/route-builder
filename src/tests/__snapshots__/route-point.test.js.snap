// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly when drag entered 1`] = `
<div
  className="route-point list__route-point list__route-point_drag-entered"
  draggable="true"
>
  <div
    className="route-point__main-info"
  >
    <span
      className="route-point__name"
    />
    <span
      className="route-point__remove-cross"
      onClick={[Function]}
      title="Remove route point"
    >
      ✖
    </span>
  </div>
  <div
    className="route-point__address-link"
  >
    View address
  </div>
  <div
    className="route-point__geo-data-info"
  />
</div>
`;

exports[`Renders correctly when drag started 1`] = `
<div
  className="route-point list__route-point list__route-point_drag-started"
  draggable="true"
>
  <div
    className="route-point__main-info"
  >
    <span
      className="route-point__name"
    />
    <span
      className="route-point__remove-cross"
      onClick={[Function]}
      title="Remove route point"
    >
      ✖
    </span>
  </div>
  <div
    className="route-point__address-link"
  >
    View address
  </div>
  <div
    className="route-point__geo-data-info"
  />
</div>
`;

exports[`Renders correctly with geo data provided 1`] = `
<div
  className="route-point list__route-point"
  draggable="true"
>
  <div
    className="route-point__main-info"
  >
    <span
      className="route-point__name"
    />
    <span
      className="route-point__remove-cross"
      onClick={[Function]}
      title="Remove route point"
    >
      ✖
    </span>
  </div>
  <div
    className="route-point__address-link"
  >
    Hide address
  </div>
  <div
    className="route-point__geo-data-info route-point__geo-data-info_shown"
  >
    name. description.
  </div>
</div>
`;

exports[`Renders correctly without geo data 1`] = `
<div
  className="route-point list__route-point"
  draggable="true"
>
  <div
    className="route-point__main-info"
  >
    <span
      className="route-point__name"
    />
    <span
      className="route-point__remove-cross"
      onClick={[Function]}
      title="Remove route point"
    >
      ✖
    </span>
  </div>
  <div
    className="route-point__address-link"
  >
    View address
  </div>
  <div
    className="route-point__geo-data-info"
  />
</div>
`;
